### ISSUE TITLE
IVar Assumption

### ISSUE THREAD

**Issue by mvz** [troessner/reek#633]
Follow-up from #483:
- Don't access a superclass' instance variables directly in a subclass.
- Don't assume presence of ivars in a module.

Generalized: Do not assume ivars are set or present outside the current class/module definition.

It may be the case that the detector for #632 will automatically also detect this one:

``` ruby
class Foo
  def initialize
    @bar = :foo
  end
end

class Bar < Foo
  def foo?
    @bar == :foo
  end
end
```

**Comment by troessner** [comment]
Cleaning up our issues: Linked this issue in #556 - closing this one.

**Comment by mvz** [comment]
But I opened all of these based on #556, so we could track them separately.

Since we'll always have more ideas than time to implement them, why not keep them open and use tags to separate urgent from non-urgent stuff?

**Comment by mvz** [comment]
Oh, no I didn't :-)

**Comment by mvz** [comment]
Second point still stands, though.

**Comment by troessner** [comment]
>> Since we'll always have more ideas than time to implement them, why not keep them open and use tags to separate urgent from non-urgent stuff?

My motivation was to cut down our open issues since we had almost 50 and for quite a few "nice to haves" it seemed like nobody was ever going to tackle those. I didn't anticipate that this was a controversial move, if I had, I would have started a quick "should we close old issues" poll.
-> Apologies for that.

>> Since we'll always have more ideas than time to implement them, why not keep them open and use tags to separate urgent from non-urgent stuff?

Yes, we can do that for sure! I'm re-opening this one and all others that are kind of "just an idea" issues.

**Comment by troessner** [comment]
@mvz I did re-open all issues I could find that were used for idea tracking, feel free to re-open anything I had missed. We should have discussed this first, sorry for that.

**Comment by mvz** [comment]
@troessner Apologies accepted, thanks! I'll have another look at old issues as well and add my comments.

**Comment by waldyr** [comment]
This is all part of a greater smell often times referred as Mystery Guest.

**Comment by mvz** [comment]
This has been implemented in #1017.
